### PR TITLE
Organization New Page & Flow

### PR DESCRIPTION
**Pull request by OAGr**
image

image

### CODE REVIEW

File: src/components/organizations/new/form.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
+
+import {create} from 'gModules/organizations/actions'
+import Card, {CardListElement} from 'gComponents/utility/card/index'
```
Comment (mmcdermott) [inline src/components/organizations/new/form.js]: Put component imports above actions, with a newline separating them. I actually find this really helpful when parsing what something is importing.

File: src/components/organizations/new/form.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+export class PlanElement extends Component {
```
Comment (mmcdermott) [inline src/components/organizations/new/form.js]: function component would be more concise here.

File: src/components/organizations/new/form.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        </div>
+        <div className='content-section'>
+          {this.props.children}
+        </div>
+      </div>
+    )
+  }
+}
+
+export class PlanList extends Component {
```
Comment (mmcdermott) [inline src/components/organizations/new/form.js]: Ditto.

File: src/components/organizations/new/form.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      </div>
+    )
+  }
+}
+
+@connect()
+export class CreateOrganizationForm extends Component {
+  state = {
+    value: '',
+    plan: 'PREMIUM'
```
Comment (mmcdermott) [inline src/components/organizations/new/form.js]: objects have trailing commas.

File: src/components/organizations/new/form.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  render() {
+    return (
+      <div className='row'>
+        <div className='col-sm-7'>
+          <div className='ui form'>
+            <div className='field name'>
+              <label>Organization Name</label>
```
Comment (mmcdermott) [inline src/components/organizations/new/form.js]: Picture URL option as well?

File: src/components/organizations/new/index.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  componentWillMount() {
+    this.props.dispatch({type: 'CLEAR_NEW_ORGANIZATION'})
+  }
+  componentWillUnmount() {
+    this.props.dispatch({type: 'CLEAR_NEW_ORGANIZATION'})
+  }
+  render() { return (<CreateOrganizationPage newOrganization={this.props.newOrganization}/>) }
+}
+
+export const CreateOrganizationPage = ({newOrganization}) => {
```
Comment (mmcdermott) [inline src/components/organizations/new/index.js]: No reason to have a separate functional component for this. It is literally directly inserted into your render above.

File: src/components/organizations/new/members.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
```
Comment (mmcdermott) [inline src/components/organizations/new/members.js]: Clean up the imports (lot of stuff here that isn't used).

File: src/components/organizations/new/members.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+import {httpRequestSelector} from 'gComponents/organizations/show/httpRequestSelector'
+import {organizationMemberSelector} from 'gComponents/organizations/show/organizationMemberSelector'
+import Card, {CardListElement} from 'gComponents/utility/card/index'
+import {MemberAddForm} from '../shared/MemberAddForm/index'
+
+import * as userOrganizationMembershipActions from 'gModules/userOrganizationMemberships/actions'
+import {organization} from 'gEngine/engine'
+
+export class LocalAddMembers extends Component {
```
Comment (mmcdermott) [inline src/components/organizations/new/members.js]: Can be functional component.

File: src/components/organizations/new/index.js (hunk 6 above)
Comment (OAGr) [inline src/components/organizations/new/index.js]: It's separate for use in the style guide.

File: src/components/organizations/shared/MemberAddForm/httpRequestSelector.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import { createSelector } from 'reselect';
```
Comment (mmcdermott) [inline src/components/organizations/shared/MemberAddForm/httpRequestSelector.js]: No semicolons, no spaces after braces in import lists.

File: src/components/organizations/shared/MemberAddForm/httpRequestSelector.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  specificHttpRequestSelector,
+  organizationIdSelector,
+  (httpRequests, organizationId) => {
+
+    let relevantRequests = httpRequests.filter(r => (
+      (r.metadata.organizationId === organizationId) &&
+      (r.entity === 'userOrganizationMembershipCreate')
+    ))
+
+    let formattedRequests = relevantRequests.map(r => ({
```
Comment (mmcdermott) [inline src/components/organizations/shared/MemberAddForm/httpRequestSelector.js]: consts. Also rename your input var to allRequests and this var to httpRequests and you can return it with {httpRequests}. Could also return inline.

File: src/components/organizations/shared/MemberAddForm/index.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+  _onKeyDown(e) {
+    if (e.keyCode === 13 && validateEmail(this.state.value)) {
+      this._submit();
+    }
+  }
+
+  render() {
+    const {value} = this.state
+    const isValid = validateEmail(value)
+    const isEmpty = _.isEmpty(value)
```
Comment (mmcdermott) [inline src/components/organizations/shared/MemberAddForm/index.js]: isEmpty should be a part of isValid IMO

File: src/components/organizations/shared/MemberAddForm/index.js [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+            })}
+      </div>
+    )
+  }
+}
+
+const InvitationHttpRequest = ({busy, success, email, isExistingMember}) => {
+  let status = httpStatus(busy, success)
+
+  if (status === 'sending'){ return (
```
Comment (mmcdermott) [inline src/components/organizations/shared/MemberAddForm/index.js]: I find this formatting very awkward.

File: src/components/organizations/show/index.js [hunk 13]
```diff
@@ -1,21 +1,24 @@
 import React, {Component, PropTypes} from 'react'
-import { connect } from 'react-redux'
+import {connect} from 'react-redux'
 
 import ReactDOM from 'react-dom'
 import Icon from 'react-fa'
 
 import {SpaceCard, NewSpaceCard} from 'gComponents/spaces/cards'
+
```
Comment (mmcdermott) [inline src/components/organizations/show/index.js]: I don't think you need some of these imports or connected selectors anymore on the org show page, right?

File: src/components/organizations/show/members.js [hunk 14]
```diff
@@ -0,0 +1,128 @@
+import React, {Component, PropTypes} from 'react'
```
Comment (mmcdermott) [inline src/components/organizations/show/members.js]: `PropTypes` import not needed

File: src/modules/organizations/actions.js [hunk 15]
```diff
@@ -38,7 +42,28 @@ export function fetchById(organizationId) {
...
+    const action = oActions.createStart(object);
+
+    api(getState()).organizations.create(object, (err, organization) => {
+      if (err) {
+        // TODO(matthew): Track if request errors out.
+        captureApiError('OrganizationsCreate', null, null, err, {url: 'OrganizationsCreate'})
+      } else if (organization) {
+        dispatch(oActions.createSuccess(organization, cid))
+        dispatch(userOrganizationMembershipActions.fetchSuccess(organization.memberships))
+        //app.router.history.navigate('/organizations/' + value.id)
```
Comment (mmcdermott) [inline src/modules/organizations/actions.js]: Get rid of this old comment.

File: src/components/organizations/new/form.js (hunk 5 above)
Comment (OAGr) [inline src/components/organizations/new/form.js]: I want this to be a separate addition.

**Pull request merged**